sct-commons/src/main/java/org/lfenergy/compas/sct/commons/util/Utils.java

```diff
@@ -313,7 +313,7 @@ public static String toHex(long number, int length) {
      * @param <T>    type of the object
      * @return copy of the object
      */
-    public static <T> T copySclElement(T object, Class<T> clazz) {
+    public static synchronized <T> T copySclElement(T object, Class<T> clazz) {
```

Comment:
I quote [JAXB user guide](https://javaee.github.io/jaxb-v2/doc/user-guide/ch03.html#other-miscellaneous-topics-performance-and-thread-safety) : 
> The JAXBContext class is thread safe, but the Marshaller, Unmarshaller, and Validator classes are not thread safe.

So it is the `unmarshaller.unmarshal` method that is not thread safe.

To allow parallel calls, you could create a new Unmarshaller every time (make the variable `unmarshaller` local), or you can synchronize the statement `unmarshaller.unmarshal` on the `unmarshaller` variable instead of synchronizing the whole method.

Reply:
Ok, Thanks